packages/intrepid2/src/Projection/Intrepid2_ProjectionToolsDefHDIV.hpp

```diff
@@ -479,12 +479,21 @@ ProjectionTools<SpT>::getHDivBasisCoeffs(Kokkos::DynRankView<basisCoeffsValueTyp
       for(ordinal_type j=0; j<numElemDofs; ++j)
         massMat(i,j) = massMat(j,i);
 
-    lapack.GESV(numElemDofs+numCurlInteriorDOFs, 1,
-        massMat.data(),
-        massMat.stride_1(),
-        (ordinal_type*)pivVec.data(),
-        rhsMat.data(),
-        rhsMat.stride_1(),
+    // lapack.GESV(numElemDofs+numCurlInteriorDOFs, 1,
+    //     massMat.data(),
+    //     massMat.stride_1(),
+    //     (ordinal_type*)pivVec.data(),
+    //     rhsMat.data(),
+    //     rhsMat.stride_1(),
+    //     &info);
+
+    // Using GELS because the matrix can be close to singular.
+    // TODO check why matrix is close to singular and possibly whether it makes sense to solve the
+    // least square problem with a better method (e.g. GGLSE)
+    lapack.GELS('N', numElemDofs+numCurlInteriorDOFs, numElemDofs+numCurlInteriorDOFs, 1,
+        massMat.data(), massMat.stride_1(),
```

Comment:
`massMat.stride(1)` works too and would be more consistent with `extent(0)` etc.